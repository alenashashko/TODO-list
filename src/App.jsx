import React, {useState, useCallback, Fragment} from "react";

import {NewTaskForm} from "./components/NewTaskForm/NewTaskForm";
import {TaskList} from "./components/TaskList/TaskList";

export const App = () => {
  const [tasks, setTasks] = useState([]);

  const handleTaskCreation = useCallback((newTask) => {
    setTasks([...tasks, newTask])
  }, [tasks]);

  const handleTaskDeletion = useCallback((taskId) => {
    setTasks(tasks.filter((task) => task.id !== taskId));
  }, [tasks]);

  return (
    <Fragment>
      <NewTaskForm onTaskCreation={handleTaskCreation} />

      <TaskList tasks={tasks} onTaskDeletion={handleTaskDeletion} />
    </Fragment>
  );
};
