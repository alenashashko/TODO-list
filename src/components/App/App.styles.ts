import { createStyles, makeStyles, Theme } from "@material-ui/core";

export const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    wrap: {
      display: "flex",
      flexDirection: "column",

      height: "100%",
    },
    mainContainer: {
      flexGrow: 1,

      paddingTop: `${theme.spacing(6)}px`,
      paddingBottom: `${theme.spacing(6)}px`,
    },
  })
);
