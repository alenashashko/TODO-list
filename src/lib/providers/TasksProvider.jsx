import React, {useContext, useState, useCallback, useMemo} from "react";
import {nanoid} from "nanoid";

const TasksContext = React.createContext({
  tasks: []
});

export const useTasksInfo = () => {
  const value = useContext(TasksContext);

  if (value === undefined) {
    throw new Error("you should use useTasksInfo inside of TasksProvider");
  }

  return value;
};

export const TasksProvider = (props) => {
  const {children} = props;

  const [tasks, setTasks] = useState([]);

  const createTask = useCallback((text) => {
    setTasks([...tasks, {
      id: nanoid(),
      date: Date.now(),
      text
    }])
  }, [tasks]);

  const changeTask = useCallback((taskId, newText) => {
    setTasks(tasks.map((task) => {
      return task.id !== taskId
        ? task
        : {
          ...task,
          date: Date.now(),
          text: newText
        }
    }))
  }, [tasks]);

  const deleteTask = useCallback((taskId) => {
    setTasks(tasks.filter((task) => task.id !== taskId));
  }, [tasks]);

  const value = useMemo(() => ({
    tasks,
    createTask,
    changeTask,
    deleteTask
  }), [tasks, createTask, changeTask, deleteTask]);

  return <TasksContext.Provider value={value}>{children}</TasksContext.Provider>;
};
