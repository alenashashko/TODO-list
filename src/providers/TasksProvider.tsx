import {FC, createContext, useContext, useState, useCallback, useMemo} from "react";
import {nanoid} from "nanoid";
import { Task } from "components/Task/Task";

const TasksContext = createContext<{
  tasks: Task[];
  createTask: (text: string) => void;
  changeTask: (taskId: string, newText: string) => void;
  deleteTask: (taskId: string) => void;
}>({
  tasks: [],
  createTask: () => {},
  changeTask: () => {},
  deleteTask: () => {},
});

export const useTasksInfo = () => {
  const value = useContext(TasksContext);

  if (value === undefined) {
    throw new Error("you should use useTasksInfo inside of TasksProvider");
  }

  return value;
};

export const TasksProvider: FC = (props) => {
  const {children} = props;

  const [tasks, setTasks] = useState<Task[]>([]);

  const createTask = useCallback((text: string) => {
    setTasks([...tasks, {
      id: nanoid(),
      date: Date.now(),
      text
    }])
  }, [tasks]);

  const changeTask = useCallback((taskId: string, newText: string) => {
    setTasks(tasks.map((task) => {
      return task.id !== taskId
        ? task
        : {
          ...task,
          date: Date.now(),
          text: newText
        }
    }))
  }, [tasks]);

  const deleteTask = useCallback((taskId: string) => {
    setTasks(tasks.filter((task) => task.id !== taskId));
  }, [tasks]);

  const value = useMemo(() => ({
    tasks,
    createTask,
    changeTask,
    deleteTask
  }), [tasks, createTask, changeTask, deleteTask]);

  return <TasksContext.Provider value={value}>{children}</TasksContext.Provider>;
};
