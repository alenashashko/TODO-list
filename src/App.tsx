import { memo } from "react";
import { Container, Grid } from "@material-ui/core";

import { Header } from "./components/Header/Header";
import { NewTaskForm } from "./components/NewTaskForm/NewTaskForm";
import { TaskList } from "./components/TaskList/TaskList";
import { Footer } from "./components/Footer/Footer";

import { useStyles } from "./App.styles";

export const App = memo(function AppView() {
  const classes = useStyles();

  return (
    <div className={classes.wrap}>
      <Header />

      <Container maxWidth="sm" className={classes.mainContainer}>
        <Grid container spacing={3}>
          <Grid item xs={12}>
            <NewTaskForm />
          </Grid>
          <Grid item xs={12}>
            <TaskList />
          </Grid>
        </Grid>
      </Container>

      <Footer />
    </div>
  );
});
