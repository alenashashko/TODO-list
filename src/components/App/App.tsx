import { memo } from "react";
import { Container, Grid } from "@material-ui/core";

import { Header } from "src/components/Header";
import { NewTaskForm } from "src/components/NewTaskForm";
import { TaskList } from "src/components/TaskList";
import { Footer } from "src/components/Footer";

import { useStyles } from "./App.styles";

export const App = memo(function AppView() {
  const classes = useStyles();

  return (
    <div className={classes.wrap}>
      <Header />

      <Container maxWidth="sm" className={classes.mainContainer}>
        <Grid container spacing={4}>
          <Grid item xs={12}>
            <NewTaskForm />
          </Grid>
          <Grid item xs={12}>
            <TaskList />
          </Grid>
        </Grid>
      </Container>

      <Footer />
    </div>
  );
});
