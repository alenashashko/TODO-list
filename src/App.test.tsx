import { render, cleanup } from "@testing-library/react";
import { App } from "./App";
import { TasksProvider } from "./providers/TasksProvider";

afterEach(cleanup);

it("renders app", () => {
  const { asFragment } = render(
    <TasksProvider>
      <App />
    </TasksProvider>
  );

  expect(asFragment()).toMatchSnapshot();
});
