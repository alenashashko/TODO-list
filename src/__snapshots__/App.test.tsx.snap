// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-wrap-1"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <h1
          class="MuiTypography-root makeStyles-title-3 MuiTypography-h6"
        >
          Todo List
        </h1>
      </div>
    </header>
    <div
      class="MuiContainer-root makeStyles-mainContainer-2 MuiContainer-maxWidthSm"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              class="MuiCardContent-root"
            >
              <h2
                class="MuiTypography-root makeStyles-title-4 MuiTypography-h5"
              >
                Create new task
              </h2>
              <form
                data-testid="form"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-flex-end"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-9"
                  >
                    <div
                      class="MuiFormControl-root MuiTextField-root makeStyles-stretch-5"
                    >
                      <div
                        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-multiline MuiInput-multiline"
                      >
                        <textarea
                          aria-invalid="false"
                          class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
                          name="task"
                          placeholder="Start typing your text here..."
                          rows="1"
                          style="height: 0px; overflow: hidden;"
                        />
                        <textarea
                          aria-hidden="true"
                          class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
                          readonly=""
                          style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); width: 100%;"
                          tabindex="-1"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-stretch-5 MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled"
                      disabled=""
                      tabindex="-1"
                      type="submit"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        Create task
                      </span>
                    </button>
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
          />
        </div>
      </div>
    </div>
    <footer
      class="MuiToolbar-root MuiToolbar-regular makeStyles-footer-6 MuiToolbar-gutters"
    >
      <h2
        class="MuiTypography-root makeStyles-copyright-7 MuiTypography-h6"
      >
        by Alena Shashko
      </h2>
    </footer>
  </div>
</DocumentFragment>
`;
